modules/cudf/src/data_frame.ts

```diff
@@ -1542,6 +1542,51 @@ export class DataFrame<T extends TypeMap = any> {
       {} as SeriesMap<{[P in keyof T]: Bool8}>));
   }
 
+  /**
+   * Compute the sum for all Series in the DataFrame.
+   *
+   * @param subset List of columns to select (all columns are considered by
+   * default).
+   * @param skipna The optional skipna if true drops NA and null values before computing reduction,
+   * else if skipna is false, reduction is computed directly.
+   * @param memoryResource Memory resource used to allocate the result Column's device memory.
+   *
+   * @returns A Series containing the sum of all values for each Series
+   * @example
+   * ```typescript
+   * import {DataFrame, Series}  from '@rapidsai/cudf';
+   *
+   * const df = new DataFrame({
+   *  a: Series.new([1, 2]),
+   *  b: Series.new([3.5, 4])
+   * });
+   * df.sum(); // [3, 7.5]
+   *
+   * const df2 = new DataFrame({
+   *  a: Series.new(['foo', 'bar']),
+   *  b: Series.new([3, 4])
+   * });
+   *
+   * df2.sum(); // throws error
+   * ```
+   */
+  sum<P extends keyof T>(subset?: (keyof T)[], skipna = true, memoryResource?: MemoryResource):
+    Series<T[P]> {
+    subset = (subset == undefined) ? this.names as (keyof T)[] : subset;
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const commonDtype = subset.reduce(
+      (type: DataType|null, name) => findCommonType(type ?? this.types[name], this.types[name]),
+      null)!;
+    const sums   = subset.map((name) => {
+      if (!(this.get(name) instanceof NumericSeries)) {
+        throw new Error('Unable to sum the results of non NumericSeries');
+      }
+      return Number(this.get(name)._col.sum(memoryResource));
```

Comment:
Would defining `sum` on `NumericSeries` help simplify this at all?

Reply:
Chatted with Bryan, seeing if there is someway to improve the `.map` type wise.

Reply:
hmm... no luck on my end. Best I came up with is currently apart of the PR.